### PR TITLE
Add reposync job

### PR DESCRIPTION
**Pull request by adammhaile**
@tbielawa @jupierce @smarterclayton 
This *should* do the trick... the doozer reposync command has already been tested out. I just need to merge this in so I can try out the job. As you'll see there's nothing here that is destructive.

### CODE REVIEW

File: jobs/build/reposync/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            stage("sync repos to local") {
+                if (IMAGES.toUpperCase() != "NONE") {
+                    command = "--working-dir ${DOOZER_WORKING} --group 'openshift-${SYNC_VERSION}' "
+                    command += "beta:reposync --output ${LOCAL_SYNC_DIR} --repo-type ${REPO_TYPE} "
+                    buildlib.doozer command
+                }
+            }
+
+            stage("push to mirror") {
+                sh "rsync -avzh --delete -e \"ssh -o StrictHostKeyChecking=no\" ${LOCAL_SYNC_DIR} ${MIRROR_TARGET}:${MIRROR_PATH} "
```
[the remark this answers is not in the document]
Comment (adammhaile) [reply jobs/build/reposync/Jenkinsfile]: Done.

**Comment by adammhaile** [comment]
@jupierce concerns addressed

File: jobs/build/reposync/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+            stage("sync repos to local") {
+                command = "--working-dir ${DOOZER_WORKING} --group 'openshift-${SYNC_VERSION}' "
+                command += "beta:reposync --output ${LOCAL_SYNC_DIR} --repo-type ${REPO_TYPE} "
+                buildlib.doozer command
+            }
+
+            stage("push to mirror") {
+                sh "rsync -avzh --delete -e \"ssh -o StrictHostKeyChecking=no\" ${LOCAL_SYNC_DIR} ${MIRROR_TARGET}:${MIRROR_PATH} "
+                buildlib.invoke_on_use_mirror("push.enterprise.sh", SYMLINK_NAME, NEW_FULL_VERSION, BUILD_MODE)
```
Comment (jupierce) [inline jobs/build/reposync/Jenkinsfile]: I don't think these arguments are defined.

File: build-scripts/use-mirror/push.enterprise.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+set -o xtrace
+
+timeout 1h /usr/local/bin/push.enterprise.sh ${REPO} -v
```
Comment (jupierce) [inline build-scripts/use-mirror/push.enterprise.sh]: Not sure how `${REPO}` would be passed in here.

**Comment by adammhaile** [comment]
Arg... Save file... THEN commit.
Will fix in the morning

**Comment by adammhaile** [comment]
fixed @jupierce

**Pull request merged**